web_s2i/application.py

```diff
@@ -1,17 +1,29 @@
-import sys
+#!/usr/bin/env python3
 
 from flask import Flask, flash, render_template, redirect, url_for, request
 from flask_mysqldb import MySQL
 from wtforms import Form, StringField, PasswordField, validators
 from passlib.hash import sha256_crypt
+import yaml
+
+
+def load_config(config_file):
+    with open(config_file, 'r') as stream:
+        try:
+            return yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+
+
+configuration = load_config('configuration.yml')
 
 application = Flask(__name__)
 
 # Config MySQL
-application.config['MYSQL_HOST'] = sys.argv[1]
-application.config['MYSQL_USER'] = sys.argv[2]
-application.config['MYSQL_PASSWORD'] = sys.argv[3]
-application.config['MYSQL_DB'] = 'devopsloft'
+application.config['MYSQL_HOST'] = configuration['mysql_host']
```

Comment:
why in the test you use environment variables and here you use the yaml?
Please be consistent, plus, the environment variables is the more standard way

Reply:
I wish I could use the yaml in the test, didn't see any Travis way of doing this - unless I will write a script to run the test and inside it will call the yaml. - what do you think ?

I always try to consider how a user in the command line will use the application, how he will be able to use the environment variables, the variables must be common.
Do you have any better reason why to move to environment variables?

Reply:
Generally, it's common to have these things passed externally via env variables, but it can be delayed for later PR

Reply:
If you have app A and database B that uses that same variable: i.e. PASSWORD
It is common to set environment variable PASSWORD and relate to it in each program?
In the command line we should add PASSWORD to .bashrc / .bash_profile or teach the user to define this variable ?

What about set of variables and much more programs? does the same rule apply ?

Reply:
If App A connects to DB B, then I don't see why not set it. Normally those variables would not be like that in env file but it's common to inject them, either via secrets injected into the pods, using tools like https://github.com/hashicorp/envconsul the variables doesn't have to be in the user's environment, more over, user's environment variables aren't passed to the containers by default unless asked to, docker however has several ways to inject them to the runtime.
When we will have more programs it will be relatively the same (you can look at my commits to my fork)
Nothing prevents us to have "overlay" of env files, and docker also supports it
https://docs.docker.com/compose/compose-file/#env_file

Again, this can always be improved in a later PR :)

Reply:
done